src/tasks/shop-admin/Rule/AssignEntityToRule.ts

```diff
@@ -0,0 +1,70 @@
+import {test as base} from '@playwright/test';
+import type {Task} from '../../../types/Task';
+import type {FixtureTypes} from '../../../types/FixtureTypes';
+import {Rule} from '../../../types/ShopwareTypes';
+
+interface Entity {
+    id: string;
+    name: string;
+    entityType: string;
+// 'shippingMethod' | 'taxProvider' | 'paymentMethod' | 'promotionOrderRule' | 'promotionCustomerRule' | 'promotionCartRule';
+}
+export const AssignEntityToRule = base.extend<{ AssignEntityToRule: Task }, FixtureTypes>({
```

Comment:
The method does more, then the name implies. Clicking on the entity to open its detail page should not be part of the assignment task. Instead it should be a separate action in your test, which starts basically the validation to check, if the correct entity has been assigned.